### PR TITLE
update docs/api-convention to v1beta3

### PR DESCRIPTION
**Pull request by caesarxuchao**
update docs/api-convention to v1beta3

@saad-ali @ghodss @nikhiljindal Could you take a look? Thanks

ref #8087

### CODE REVIEW

File: docs/api-conventions.md [hunk 1]
```diff
@@ -462,24 +462,24 @@ The status object is encoded as JSON and provided as the body of the response.
 **Example:**
 ```JSON
 >HTTP Requst:
-POST /api/v1beta1/events/ HTTP/1.1
-Authorization: Basic ...
+>POST /api/v1beta3/namespaces/default/events/ HTTP/1.1
```
Comment (saad-ali) [inline docs/api-conventions.md]: While you're at it, might as well use a better example:
```
> GET /api/v1beta3/namespaces/default/pods/grafana HTTP/1.1
 
< HTTP/1.1 404 Not Found
< Content-Type: application/json
< Date: Wed, 20 May 2015 00:21:46 GMT
< Content-Length: 232
 
{
  "kind": "Status",
  "apiVersion": "v1beta3",
  "metadata": {},
  "status": "Failure",
  "message": "pods \"grafana\" not found",
  "reason": "NotFound",
  "details": {
    "id": "grafana",
    "kind": "pods"
  },
  "code": 404
}
```
This shows the optional `details` field as well.

Comment (caesarxuchao) [inline docs/api-conventions.md]: Thanks @saad-ali. I've updated the example. Could you take another look?

**Comment by saad-ali** [comment]
Why all the the changes to lower case? LGTM otherwise. Once you change that back, please squash your commits.

**Comment by caesarxuchao** [comment]
Done. Please take another look. Thanks. @saad-ali

**Comment by saad-ali** [comment]
LGTM. Ok to merge, since it's only doc changes.

**Pull request merged**